src/Behat/Behat/Output/ServiceContainer/Formatter/JUnitFormatterFactory.php

```diff
@@ -0,0 +1,156 @@
+<?php
+
+/*
+ * This file is part of the Behat.
+ * (c) Konstantin Kudryashov <ever.zet@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Behat\Behat\Output\ServiceContainer\Formatter;
+
+use Behat\Testwork\Output\ServiceContainer\OutputExtension;
+use Behat\Testwork\Exception\ServiceContainer\ExceptionExtension;
+use Behat\Testwork\Output\ServiceContainer\Formatter\FormatterFactory;
+use Symfony\Component\DependencyInjection\Reference;
+use Symfony\Component\DependencyInjection\Definition;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+
+/**
+ * Behat junit formatter factory.
+ *
+ * @author Wouter J <wouter@wouterj.nl>
+ */
+class JUnitFormatterFactory implements FormatterFactory
```

Comment:
It's much easier to be more restrictive initially than to bring in restrictions later. Making the class final (if there are no abstract elements to it) means that it can't be overridden, which helps towards the overall best practice of favouring composition over inheritance, as I understand it anyway.